experimental/personnames/src/derive_missing_initials.rs

```diff
@@ -0,0 +1,170 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use lazy_static::lazy_static;
+use regex::CaptureMatches;
+use regex::Regex;
+
+use crate::api::{FieldLength, FieldModifier, NameField, PersonName};
+
+const CLDR_INITIAL_INITIAL_PATTERN: &str = r"\{(?P<initial_position>\d+)}(?P<trailing>[^{]+)?";
+const CLDR_INITIAL_SEQUENCE_PATTERN: &str = r"\{(?P<initial_position>\d+)}(?P<trailing>[^{]+)?";
```

Comment:
Question: do these regexes come from the spec?

Probably okay to hard-code them but we should consider using regex-automata. But what you have written should be fine to start. It's something we should look into before stabilization.